pkg/api/vlabs/types.go

```diff
@@ -184,11 +184,22 @@ type KubernetesContainerSpec struct {
 	MemoryLimits   string `json:"memoryLimits,omitempty"`
 }
 
+// KubernetesEnvironmentSpec defines configuration for environment variables spec for ACI Connector
+type KubernetesEnvironmentSpec struct {
+	ClientID       string `json:"clientid,omitempty"`
+	ClientKey      string `json:"clientkey,omitempty"`
+	TenantID       string `json:"tenantid,omitempty"`
+	SubscriptionID string `json:"subscriptionid,omitempty"`
+	ResourceGroup  string `json:"resourcegroup,omitempty"`
+	Region         string `json:"region,omitempty"`
+}
+
 // KubernetesAddon defines a list of addons w/ configuration to include with the cluster deployment
 type KubernetesAddon struct {
-	Name       string                    `json:"name,omitempty"`
-	Enabled    *bool                     `json:"enabled,omitempty"`
-	Containers []KubernetesContainerSpec `json:"containers,omitempty"`
+	Name        string                      `json:"name,omitempty"`
+	Enabled     *bool                       `json:"enabled,omitempty"`
+	Containers  []KubernetesContainerSpec   `json:"containers,omitempty"`
+	Environment []KubernetesEnvironmentSpec `json:"environment,omitempty"`
```

Comment:
It would be great to make this a `map[string]string` so it has general purpose value. `KubernetesAddon` is meant to be a generic interface for configuring an addon.

Reply:
How about `Config` map[string]string` `json:"config,omitempty"`